### PR TITLE
tailfs: listen for local clients only on 100.100.100.100

### PR DESCRIPTION
**Pull request by oxtoacart**
FileSystemForLocal was listening on the node's Tailscale address, which potentially exposes the user's view of TailFS shares to other Tailnet users. Remote nodes should connect to exported shares via the peerapi.

This removes that code so that FileSystemForLocal is only avaialable on 100.100.100.100:8080.

Updates tailscale/corp#16827

### CODE REVIEW

File: wgengine/netstack/netstack.go [hunk 1]
```diff
@@ -919,10 +919,10 @@ func (ns *Impl) acceptTCP(r *tcp.ForwarderRequest) {
 		return gonet.NewTCPConn(&wq, ep)
 	}
 
-	// Local DNS Service (DNS and WebDAV)
+	// Local Services (DNS and WebDAV)
 	hittingServiceIP := dialIP == serviceIP || dialIP == serviceIPv6
 	hittingDNS := hittingServiceIP && reqDetails.LocalPort == 53
-	hittingTailFS := hittingServiceIP && ns.tailFSForLocal != nil && reqDetails.LocalPort == 8080
+	hittingTailFS := hittingServiceIP && ns.tailFSForLocal != nil && reqDetails.LocalPort == ipnlocal.TailFSLocalPort
```
Comment (oxtoacart) [inline wgengine/netstack/netstack.go:925]: Drive-by, might as well use the constant.

**Review by catzkorn — COMMENTED** [review]
Looks okay to me, will close the issue!
